crates/ruff_python_index/src/indexer.rs

```diff
@@ -74,9 +84,15 @@ impl Indexer {
             continuation_lines,
             fstring_ranges: fstring_ranges_builder.finish(),
             multiline_ranges: multiline_ranges_builder.finish(),
+            comment_ranges: CommentRanges::new(comment_ranges),
```

Comment:
I think the advantage of the `CommentsBuilder` was to enforce that the tokens are guaranteed to be in sorted order.  This is now no longer guaranteed where `CommentRange`s can be built by "anyone". 

I would probably keep the builder and have two methods:

```
CommentRangesBuilder::default().from_tokens(tokens)

// or

CommentRangesBuilder::default().push_token(token).push_token(token...).finish()
```

Reply:
I think we could add it as a debug assertion

Reply:
Not sure what's the advantage of `CommentRangesBuilder.from_tokens` is because `Tokens` is guaranteed in sorted order otherwise everything would just break 😅

Reply:
That's true. Lol, I should not make things complicated. Maybe keep what you had